server/inventory_item/updater.py

```diff
@@ -0,0 +1,8 @@
+from apscheduler.schedulers.background import BackgroundScheduler
+from django_server.load_csv_to_db import main
+
+
+def start():
+    scheduler = BackgroundScheduler()
+    scheduler.add_job(main, 'interval', minutes=1)
```

Comment:
Is this starting a thread or a process? 
Can you explain how a job is implemented in this library?

Reply:
APScheduler has four kinds of components:

- triggers

- job stores

- executors

- schedulers

Triggers contain the scheduling logic. Each job has its own trigger which determines when the job should be run next. Beyond their initial configuration, triggers are completely stateless.

Job stores house the scheduled jobs. The default job store simply keeps the jobs in memory, but others store them in various kinds of databases. A job’s data is serialized when it is saved to a persistent job store, and deserialized when it’s loaded back from it. Job stores (other than the default one) don’t keep the job data in memory, but act as middlemen for saving, loading, updating and searching jobs in the backend. Job stores must never be shared between schedulers.

Executors are what handle the running of the jobs. They do this typically by submitting the designated callable in a job to a thread or process pool. When the job is done, the executor notifies the scheduler which then emits an appropriate event.

Schedulers are what bind the rest together. You typically have only one scheduler running in your application. The application developer doesn’t normally deal with the job stores, executors or triggers directly. Instead, the scheduler provides the proper interface to handle all those. Configuring the job stores and executors is done through the scheduler, as is adding, modifying and removing jobs.

Reply:
I get that, but what about the job itself? How is it implemented? What is the underlying python process?

When you start a job, what is the library doing? We need to know this before using the library, because if we have cpu-bound tasks, we can't be relying on this if it's using threads.

Reply:
This can be configured. "The default ThreadPoolExecutor should be good enough for most purposes. If your workload involves CPU intensive operations, you should consider using ProcessPoolExecutor instead to make use of multiple CPU cores. You could even use both at once, adding the process pool executor as a secondary executor."

Reply:
okay, awesome, this is what I wanted to know.

Reply:
> Problem 1: I'm starting to see the direction where we're going with this code right now, and the current design does not easily allow for updating inventory items from multiple organizations. We could separate this out into another dev story for a separate sprint, since Dominik is also probably going to be relying on this design as well.
> 
> What are your thoughts? where should we go from here for this problem?


Problem 1:
Okay, I see what you mean, but one thing I don't really understand is that the database won't be a centralized one for all the organizations. Can you clarify a bit? Because as of now we just have one inventory items collection, how are we suppose to handle that? Maybe we could discuss it on a call.